dll/ntdll/csr/connect.c

```diff
@@ -366,11 +371,18 @@ CsrClientCallServer(IN OUT PCSR_API_MESSAGE ApiMessage,
     ULONG PointerCount;
     PULONG_PTR OffsetPointer;
 
+    /* Make sure the length is valid */
+    if (DataLength > (MAXSHORT - sizeof(CSR_API_MESSAGE)))
+    {
+        DPRINT1("DataLength too big: %lu", DataLength);
+        return STATUS_INVALID_PARAMETER;
+    }
+
     /* Fill out the Port Message Header */
     ApiMessage->Header.u2.ZeroInit = 0;
-    ApiMessage->Header.u1.s1.TotalLength = DataLength +
-        sizeof(CSR_API_MESSAGE) - sizeof(ApiMessage->Data); // FIELD_OFFSET(CSR_API_MESSAGE, Data) + DataLength;
-    ApiMessage->Header.u1.s1.DataLength = DataLength +
+    ApiMessage->Header.u1.s1.TotalLength = (CSHORT)DataLength +
+        sizeof(CSR_API_MESSAGE) - sizeof(ApiMessage->Data); // FIELD_OFFSET(CSR_API_MESSAGE, Data) + DataLength;;
```

Comment:
Please remove the extra trailing semicolon that has been added at the end of the comment.

Reply:
done